src/main/java/com/google/graphgeckos/dashboard/storage/DataRepository.java

```diff
@@ -80,4 +81,26 @@
    * @throws IllegalArgumentException if commitHash is null
    */
   BuildInfo getRevisionEntry(String commitHash) throws IllegalArgumentException;
+
+  /**
+   * Queries the database for the last known index for a specific buildbot. Used by the fetchers.
+   *
+   * @param buildbotName the name of the buildbot to search for
+   * @return the last known revision index for which there is information in the database.
+   * @throws IllegalArgumentException if {@code buildbotName} is null
+   */
+  int getBuildbotIndex(String buildbotName) throws IllegalArgumentException, NotBoundException;
```

Comment:
Remove the `IllegalArgumentException` from the throws block. `IllegalArgumentException` is a runtime exception. Declaring it in the throws block has no meaning in Java. See https://www.geeksforgeeks.org/checked-vs-unchecked-exceptions-in-java/

Roughly speaking, you'd use checked exceptions for cases where an error is expected and your code needs to react to it. Unchecked exceptions like `IllegalArgumentException` are used to signal bugs. For example, it would be a bug in you application if `getBuildbotIndex(null)` gets called.

Reply:
In this case, I believe the code needs to react to the cases in which these exceptions would have been thrown. Even if some bug occurs where the argument is null, the server should not stop. Should this exception handling be trusted to the user, and not enforced?

Reply:
@MagoDelBlocco, there is no difference between having runtime exceptions in the `throws` clause or not. I think it is acceptable to have them there for documentation purposes, but considering that you also have it documented in the JavaDoc, this is of limited value.

Java servers at Google usually handle runtime exceptions by letting them bubble up the whole request and catching them there. That way the server does not die, but the request error handling needs to be done only in one place, typically by returning a 5xx error code.

Reply:
I have removed the `throws` clause for all methods, while maintaining the JavaDoc documentation about the exceptions. Thank you for the explanation!